File: 0088-merge-sorted-array/0088-merge-sorted-array.cpp
class Solution {
public:
    void merge(vector<int>& nums1, int m, vector<int>& nums2, int n) {
        vector<int> temp (m + n, 0);
        int first_index = 0, second_index = 0, temp_index = 0;      
        while (first_index <= m - 1 && second_index <= n - 1) 
        {
            if (nums1[first_index] <= nums2[second_index]) 
            {   temp[temp_index] = nums1[first_index];
                temp_index++;
                first_index++;
            }
            else
            {
                temp[temp_index] = nums2[second_index];
                temp_index++;
                second_index++; 
            }
        }
        
        while (first_index <= m - 1)
        {
            temp[temp_index] = nums1[first_index];
            first_index++;
            temp_index++;
        }
        while (second_index <= n - 1)
        {
            temp[temp_index] = nums2[second_index];
            second_index++;
            temp_index++;
        }
        for (int index = 0; index < m + n; index++)
        {
            nums1[index] = temp[index];
        }
    }
};

File: 0088-merge-sorted-array/0088-merge-sorted-array.py
class Solution:
    def merge(self, nums1: List[int], m: int, nums2: List[int], n: int) -> None:
        """
        Do not return anything, modify nums1 in-place instead.
        """
        if not nums2:
            return 
        temp = [0 for _ in range(m + n)]
        first_index, second_index, temp_index = 0, 0, 0         
        while first_index <= m - 1 and second_index <= n - 1:
            if nums1[first_index] <= nums2[second_index]:
                temp[temp_index] = nums1[first_index]
                temp_index += 1
                first_index += 1
            else:
                temp[temp_index] = nums2[second_index]
                temp_index += 1
                second_index += 1
        
        while first_index <= m - 1:
            temp[temp_index] = nums1[first_index]
            first_index += 1
            temp_index += 1
        while second_index <= n - 1:
            temp[temp_index] = nums2[second_index]
            second_index += 1
            temp_index += 1
        
        for index in range(m + n):
            nums1[index] = temp[index]
            
                